anaconda_lib/linting/anaconda_mypy.py

```diff
@@ -63,48 +50,31 @@ def execute(self):
         return errors
 
     def check_source(self):
-        """Wrap calls to MyPy as a library
-        """
 
-        err_ctx = '--hide-error-context'
-        if MYPY_VERSION < (0, 4, 5):
-            err_ctx = '--suppress-error-context'
+        forced_args = ['--hide-error-context', '--show-traceback']
+        all_args = self.settings + forced_args + [self.filename]
 
-        args = shlex.split('\'{0}\' -O -m mypy {1} {2} \'{3}\''.format(
-            sys.executable, err_ctx,
-            ' '.join(self.settings[:-1]), self.filename)
-        )
-        env = os.environ.copy()
         if self.mypypath is not None and self.mypypath != "":
-            env['MYPYPATH'] = self.mypypath
-
-        kwargs = {
-            'cwd': os.path.dirname(os.path.abspath(__file__)),
-            'bufsize': -1,
-            'env': env
-        }
-        if os.name == 'nt':
-            startupinfo = subprocess.STARTUPINFO()
-            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
-            kwargs['startupinfo'] = startupinfo
-
-        proc = Popen(args, stdout=PIPE, stderr=PIPE, **kwargs)
-        out, err = proc.communicate()
+            os.environ['MYPYPATH'] = self.mypypath
+
+        logging.info('calling mypy with %s' % str(all_args))
+
+        (out, err, status) = mypyApi.run(all_args)
+		
         if err is not None and len(err) > 0:
-            if sys.version_info >= (3,):
-                err = err.decode('utf8')
             raise RuntimeError(err)
 
-        if sys.version_info >= (3,):
-            out = out.decode('utf8')
-
         errors = []
         for line in out.splitlines():
-            if (self.settings[-1] and not
-                    self.silent and 'stub' in line.lower()):
-                continue
 
             data = line.split(':') if os.name != 'nt' else line[2:].split(':')
+
+            # mypy has paths relative to project, anaconda has paths absolute paths.
+            # mypy returns errors that are not necessarily just for the current file.
+            filename = data[0]
+            if not self.filename.endswith(filename):
+                continue
+
             errors.append({
```

Comment:
Please wrap this in a try block.  Else it will fail in .7.5.0 when it hits the line `Found 1 error in 1 file (checked 1 source file)` and display no errors.  
```
try:
    errors.append({
        'level': 'W',
        'lineno': int(data[1]),
        'offset': 0,
        'code': ' ',
        'raw_error': '[W] MyPy {0}: {1}'.format(
            data[2], data[3]
        ),
        'message': '[W] MyPy%s: %s',
        'underline_range': True
    })
except IndexError:
    pass
```